File: linked_list/cxx/LinkedList.cxx
#include "LinkedList.h"

ListNode* getLinkedList(std::vector<int> array)
{
	if(array.empty())
		return NULL;

	ListNode* head = new ListNode(array[0]);
	ListNode* curr = head;
	for(int i = 1; i < array.size(); i++)
	{
		curr->next = new ListNode(array[i]);
		curr = curr->next;
	}
	return head;
}

std::vector<int> getArray(ListNode* head)
{
	std::vector<int> array;
	ListNode* curr = head;
	while(curr)
	{
		array.push_back(curr->val);
		curr = curr->next;
	}
	return array;
}

void destroyLinkedList(ListNode* head)
{
	while(head != NULL)
	{
		delete head;
		head = head->next;
	}
}

File: linked_list/py3/LinkedList.py
class ListNode(object):
    def __init__(self, x):
        self.val = x
        self.next = None

def getLinkedList(array):
    if not array:
        return None
    curr = head = ListNode(array[0])
    for i in range(1, len(array)):
        curr.next = ListNode(array[i])
        curr = curr.next
    return head

def getArray(head):
    curr = head
    array = []
    while curr:
        array.append(curr.val)
        curr = curr.next
    return array
